Source/src/modules/ModuleRender.cpp

```diff
@@ -365,35 +366,64 @@ void Hachiko::ModuleRender::DrawDeferred(Scene* scene, ComponentCamera* camera,
     batch_manager->ClearTransparentBatchesDrawList();
 
     // If forward pass is disabled on the settings, return:
-    if (!render_forward_pass)
+    if (render_forward_pass)
     {
-        return;
-    }
+        // Get transparent meshes:
+        const std::vector<RenderTarget>& transparent_targets = render_list.GetTransparentTargets();
 
-    // Get transparent meshes:
-    const std::vector<RenderTarget>& transparent_targets = render_list.GetTransparentTargets();
+        if (transparent_targets.size() > 0)
+        {
+            // Get the targets that has transparent materials. These targets will be
+            // rendered with regular forward rendering pass:
+            for (const RenderTarget& target : transparent_targets)
+            {
+                batch_manager->AddDrawComponent(target.mesh_renderer);
+            }
 
-    if (transparent_targets.size() <= 0)
-    {
-        return;
+            // Forward rendering pass for transparent game objects:
+            program = App->program->GetForwardProgram();
+            program->Activate();
+
+            EnableBlending();
+            batch_manager->DrawTransparentBatches(program);
+            DisableBlending();
+
+            g_buffer.BindForDrawing();
+            // Forward depth (Used for fog)
+            program = App->program->GetTransparentDepthProgram();
+            program->Activate();
+                
+            batch_manager->DrawTransparentBatches(program);
+            Program::Deactivate();
+            glBindFramebuffer(GL_DRAW_FRAMEBUFFER, frame_buffer);
+        }
     }
-    
-    // Get the targets that has transparent materials. These targets will be
-    // rendered with regular forward rendering pass:
-    for (const RenderTarget& target : transparent_targets)
+
+    // ----------------------------- FOG -----------------------------
+
+    if (scene->IsFogEnabled())
     {
-        batch_manager->AddDrawComponent(target.mesh_renderer);
-    }
+        // Bind all g-buffer textures:
+        g_buffer.BindForReading();
+        g_buffer.BindFogTextures();
+        
+        program = App->program->GetFogProgram();
+        program->Activate();
 
-    // Forward rendering pass for transparent game objects:
-    program = App->program->GetForwardProgram();
-    program->Activate();
+        EnableBlending();
+        
+        program->BindUniformFloat3("fog_color", scene->GetFogColor().ptr());
+        program->BindUniformFloat("fog_global_density", scene->GetFogGlobalDensity());
+        program->BindUniformFloat("fog_height_falloff", scene->GetFogHeightFalloff());
```

Comment:
See Uniforms.h for these type of uniform names, across the engine, we are now trying to gather all uniform names in a global space and use those instead of these "magic strings". On Uniforms.h you can add a namespace inside that is called Fog, and add these uniform binding names there as constexprs.